### PR TITLE
[mono][llvm] Refactor code to make it easier to support LLVM 13.x in …

### PR DESCRIPTION
**Pull request by vargaz**
…the future

* Remove some c++ functions which can be implemented using the C API.
* Use call_intrins in more places.

### CODE REVIEW

**Review by lateralusX — APPROVED** [review]

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm.c [hunk 1]
```diff
@@ -2562,51 +2574,48 @@ emit_cond_system_exception (EmitContext *ctx, MonoBasicBlock *bb, const char *ex
 		return;
 	}
 
-	callee = ctx->module->throw_corlib_exception;
-	if (!callee) {
```
Comment (fanyang-mono) [inline src/mono/mono/mini/mini-llvm.c:2566]: Why removing this check?

Comment (vargaz) [reply src/mono/mono/mini/mini-llvm.c:2566]: These were caches, but they were not updated so they never worked.

Comment (fanyang-mono) [reply src/mono/mono/mini/mini-llvm.c:2566]: Gotcha!

**Review by fanyang-mono — APPROVED** [review]

**Pull request merged**